pic32/libraries/DTWI/DTWI.cpp

```diff
@@ -926,6 +926,118 @@ bool DTWI::stopMaster(void)
+ *
+ *    Description: 
+ *      Converts a frequency in Hz to an I2C_FREQ enum.
+ *
+ *      Unfortunately, due to beginMaster() using a I2C_FREQ enum vs
+ *      an actual frequency, the mapping of frequencies to enums
+ *      is slightly different dependind on picking an enum based on an
+ *      arbitrary frequency or picking an enum from an actual frequency
+ *      that was previously created from an enum.
+ *
+ *	When picking an enum from an arbitrary frequency, round down
+ *      which ensures that the frequency enum will be no greater than the
+ *      desired frequency.
+ *
+ *      When picking an enum from an actual frequency, round up
+ *      to ensure that beginMaster() will create the same actual
+ *      frequency.
+ *
+ *	If freqHz is a frequency that is exactly supported,
+ *	the rounding direction does not matter.
+ *       
+ * ------------------------------------------------------------ */
+
+DTWI::I2C_FREQ DTWI::freqHz2i2c_freq(uint32_t freqHz, int round)
+{
+    // round up or down?
+    if(round < 0)
+    {
+        // pick closest frequency enum without going over
+        if(freqHz < FQ10KHz)  return(FQ0Hz);
+        if(freqHz < FQ50KHz)  return(FQ10KHz);
+        if(freqHz < FQ100KHz) return(FQ50KHz);
+        if(freqHz < FQ400KHz) return(FQ100KHz);
+        if(freqHz < FQ1MHz)   return(FQ400KHz);
+        return(FQ1MHz);
+    }
+    else
+    {
+        // pick closest I2C_FREQ enum that should give same actual frequency
+        // when used with beginMaster()
+        // note: this assumes freqHz was an actual frequency created using
+        // the I2C_FREQ enums.
+        if(!freqHz)             return(FQ0Hz);
+        if(freqHz <=  FQ10KHz)  return(FQ10KHz);
+        if(freqHz <=  FQ50KHz)  return(FQ50KHz);
+        if(freqHz <=  FQ100KHz) return(FQ100KHz);
+        if(freqHz <= FQ400KHz)  return(FQ400KHz);
+        return(FQ1MHz);
+    }
+}
+
+/***    I2C_FREQ getClock(uint32_t *pfreqHz = NULL)
+ *
+ *    Parameters:
+ *          pfreqHz:
+ *            optional and if provided and non NULL,
+ *            will point to the location where to store actual i2c
+ *            master clock frequency in Hz.
+ *              
+ *    Return Values:
+ *          The current I2C_FREQ frequency of the i2c clock being used.
+ *          If not in master mode or not initilized FQ0Hz is returned.
+ *	
+ *
+ *    Description: 
+ *	Checks for master mode and intialization.
+ *      If h/w is in master mode and initalized, it reads the clock register
+ *      and back calculates the actual frequency.
+ *	Then, it must convert the actual frequency back to a I2C_FREQ enum.
+ *	getClock() must round up from the actual frequency to the nearest enum
+ *      value to enusure that the same clock frequency is created from the
+ *      enum when handed to beginMaster()
+ *
+ * ------------------------------------------------------------ */
+DTWI::I2C_FREQ DTWI::getClock(uint32_t *pfreqHz)
+{
+uint32_t freqHz = 0;
+I2C_FREQ I2Cfreq;
```

Comment:
Indenting on these two lines : should be 2 spaces before the variable declarations so they line up with the if( below.

Reply:
That module actually uses 4 spaces of indention.

Reply:
Ahh, right-o. Then let's go with that-